src/config.c

```diff
@@ -73,43 +74,124 @@ static bool session_access_level_check(int *newval, void **extra, GucSource sour
   return true;
 }
 
-/* `check_hook`s for intervals only issue warnings, so they only make sense in interactive mode. */
-static bool outlier_count_min_check_hook(int *newval, void **extra, GucSource source)
+static const double MIN_STRICT_NOISE_LAYER_SD = 1.0;
+static const int MIN_STRICT_LOW_COUNT_MIN_THRESHOLD = 2;
+static const double MIN_STRICT_LOW_COUNT_MEAN_GAP = 2.0;
+static const double MIN_STRICT_LOW_COUNT_LAYER_SD = 1.0;
+static const int MIN_STRICT_OUTLIER_COUNT_MIN = 1;
+static const int MIN_STRICT_OUTLIER_COUNT_MAX = 2;
+static const int MIN_STRICT_TOP_COUNT_MIN = 2;
+static const int MIN_STRICT_TOP_COUNT_MAX = 3;
+static const int MIN_STRICT_INTERVAL_SIZE = 1;
+
+static bool strict_check_hook(bool *newval, void **extra, GucSource source)
 {
-  if (source >= PGC_S_INTERACTIVE && *newval > g_config.outlier_count_max)
+  if (source > PGC_S_DYNAMIC_DEFAULT && *newval)
   {
-    NOTICE_LOG("Outlier count interval invalid: (%d, %d). Set upper bound to make it valid.", *newval, g_config.outlier_count_max);
+    bool incorrect = g_config.noise_layer_sd < MIN_STRICT_NOISE_LAYER_SD ||
+                     g_config.low_count_min_threshold < MIN_STRICT_LOW_COUNT_MIN_THRESHOLD ||
+                     g_config.low_count_mean_gap < MIN_STRICT_LOW_COUNT_MEAN_GAP ||
+                     g_config.low_count_layer_sd < MIN_STRICT_LOW_COUNT_LAYER_SD ||
+                     g_config.outlier_count_min < MIN_STRICT_OUTLIER_COUNT_MIN ||
+                     g_config.outlier_count_max < MIN_STRICT_OUTLIER_COUNT_MAX ||
+                     g_config.top_count_min < MIN_STRICT_TOP_COUNT_MIN ||
+                     g_config.top_count_max < MIN_STRICT_TOP_COUNT_MAX ||
+                     g_config.outlier_count_max - g_config.outlier_count_min < MIN_STRICT_INTERVAL_SIZE ||
+                     g_config.top_count_max - g_config.top_count_min < MIN_STRICT_INTERVAL_SIZE;
+    if (incorrect)
+    {
+      NOTICE_LOG("Current values of anonymization parameters not conforming to pg_diffix.strict = true.");
```

Comment:
The last part is a little confusing.

```suggestion
      NOTICE_LOG("Current values of anonymization parameters not conforming to strict mode.");
```